Automattic/jetpack
# CRM: Enable JS linting

Linting has long been disabled on CRM JS files (for good reason: ensuring compliance required a lot of work).

This PR enables linting and cleans up the JS files.

I tried to be careful in my changes, but now it's your turn to review. Might be worth splitting the task between multiple people. Some things of note:

* In several cases we were setting and exporting a global-scope variable, but then referring to it with `window.*`. To prevent lint change cascades, I removed the export and simply switched them to be initialized as `window.*`. Most importantly, make sure I didn't set any top-level constants.
* I tried to be careful with strict comparison. In a few cases it was safe to make it more strict, but in others we can't without more digging. Make sure any changes made to those are acceptable. 
* Due to the sheer quantity of JSDoc changes needed and the legwork needed to ensure proper documentation, several large files have those rules disabled on the file level at this time.
* There was a beta feedback JS file, and we don't use any of that related code, so I chopped that and its related code: 5154cd40b87160136d04e44d610314b919b2e353

Good luck! On the bright side, I deliberately didn't put other chops I found in this branch; another one is forthcoming. 😅 

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






1. Click through the CRM and make sure there are no new errors.
2. Check the code (probably with whitespace disabled).
